pkg/datastore/pubsub/google/google.go

```diff
@@ -133,6 +133,12 @@ func createTopicsIfNotExist(ctx context.Context, g *GCPubSub) error {
 }
 
 func createSubscription(ctx context.Context, g *GCPubSub) error {
+
+	// Check if SubscriptionDetails is nil before accessing its properties
+	if g.config.SubscriptionDetails == nil {
```

Comment:
Now, this is more open to improvement!

The question is do we always require a subscription?

If the user comes looking to create a publish-only application does google pubsub support publishing without a subscription?

If yes, we should skip creating a subscription when no subscription details are provided, and a check should be put when a user is trying to subscribe with no subscriptions!

Reply:
Google Cloud PubSub is designed to ensure that messages are delivered to subscribers at least once. This means that even in scenarios where a subscriber is temporarily unavailable, PubSub retains unacknowledged messages in persistent storage. These messages are stored for up to 7 days from their publication date.

Notably, messages that are published before the creation of a subscription cannot be retrieved later. Additionally, there's a maximum message retention period of 7 days. After this period, any undelivered messages are permanently deleted.

Therefore, it is a standard behavior in Google Cloud PubSub for subscribers to receive messages that were published while they were inactive, as long as the messages are less than 7 days old. This feature is particularly useful for ensuring message delivery continuity, even during temporary subscriber downtime.